internal/pkg/gateway/api.go

```diff
@@ -273,6 +270,44 @@ func (gs *Server) Endorse(ctx context.Context, request *gp.EndorseRequest) (*gp.
 	return endorseResponse, nil
 }
 
+// responseStatus unpacks the proposal response and error values that are returned from ProcessProposal and
+// determines how the gateway should react (retry?, close connection?).
+// Uses the grpc canonical status error codes and their recommended actions.
+// Returns:
+// - response successful (bool)
+// - error message extracted from the err or generated from 500 proposal response
+// - should the gateway retry (only the Evaluate() uses this)
```

Comment:
Why does only Evaluate use this?

Reply:
Ok, I'd suggest summarize the retry logic in the Endorse() and Evaluate() function comments.

Reply:
Comments that duplicate the logic of the code are a well-known bad practice. They add clutter/duplication and tend to get out of date and become misleading as the code changes. Could we focus less on accurately describing the code in comments, and focus more on clearly expressing the logic in code?

Reply:
@bestbeforetoday  Of course the priority should be on expressive logic in code.
I'm talking about a top level function comment on Endorse() and Evaluate(). Retry is such a fundamental and non-obvious aspect of this top-level public contract that it should at least be mentioned in the function comment that serves as the contract documentation.
See the example from the Go authors - https://go.dev/blog/godoc

Reply:
Endorse and Evaluate are gRPC services. Their documentation comments should be in the protobuf definitions, not in implementation code.

Documentation should describe the external behaviour the service (or function) provides, not the implementation details of how it provides that service.

Reply:
Retry is certainly something clients need to know that the Gateway services are doing, otherwise they will attempt their own retry strategies that overlap with what Gateway does. Agree that it would go best in the protobuf definitions.